### ISSUE TITLE
docs: invalid version number in `Working with Salesforce` section

### ISSUE THREAD

**Issue by techfg** [skuid/skuid-cli#206]
In the updates that were made to address #183 and #204, the docs were adjusted to state `...no longer supported as of v6 of the CLI`, however there is no `v6` of the cli.

In reviewing the commit history, the best that I can tell is the following:

1. `SFX` appears to have been removed entirely in [this commit](https://github.com/skuid/skuid-cli/commit/ca75f73a8774d13af9322cf4f8885f797876e195) on 2023/05/30 11:29AM PDT.  Of interest here is that this commit actually seems to come from #87 which indicates it was merged to `master` on 2022/09/06 3:05pm PDT, 8 months prior to what ca75f73a8774d13af9322cf4f8885f797876e195 currently reflects in `master` as the commit date.
2. The above commit changed the version of the CLI to [v1.0.0](https://github.com/skuid/skuid-cli/commit/ca75f73a8774d13af9322cf4f8885f797876e195#diff-281bb08d90b580496d51ee4a4e13554becf791405392222f16531ffa3e80a414R1).
3. This [commit](https://github.com/skuid/skuid-cli/commit/e302ff07c44f509965adbd9adef9cc3874e8242c#diff-281bb08d90b580496d51ee4a4e13554becf791405392222f16531ffa3e80a414R1) changed the version to `0.6.0` on 2023/05/30 11:31AM PDT.  Of interest here beyond the version change is that the commit appears to come from #99 which was merged to `master` on 2023.01.19 1:47pm PDT, 4 months prior to what 281bb08d90b580496d51ee4a4e13554becf791405392222f16531ffa3e80a414R1 currently reflects in `master` as the commit date.

Assuming that the `v6` mentioned in the docs was intended to be `v.0.6.0` and the incorrect reference just wasn't caught during doc update and/or doc review (assuming there was one), however even `v0.6.0` is not 100% correct if you correlate the items in the [Releases](https://github.com/skuid/skuid-cli/releases) page.

The `Releases` page contains `alpha` releases for `0.6.0-alpha`, `0.6.1-alpha`, `0.6.2-alpha`, `0.6.3-alpha` and `0.6.4-alpha` with the first GA release of `0.6.x` being `0.6.5`.  Unfortunately, the releases containing the name `alpha` did not contain `alpha` in their semver.  For example, if you run the binary for [0.6.0-alpha](https://github.com/skuid/skuid-cli/releases/tag/0.6.0-alpha), the version displayed in `0.6.0` and the code tagged with [0.6.0-alpha](https://github.com/skuid/skuid-cli/tree/0.6.0-alpha) contains a [0.6.0](https://github.com/skuid/skuid-cli/blob/0.6.0-alpha/.version#L1) version reference.  

The result of this is that even stating that `sfx` is no longer supported as of `v0.6.0` is inaccurate as it relates to the `named releases` on the `Releases` page since semver wise, `v0.6.0-alpha` comes before `v0.6.0`.

In short, it's not entirely clear what version should be considered the actual version `sfx` support was removed entirely so will defer to Skuid to make the final determination.  Possibly updating the docs to be `v0.6.x` would cover all the above, or changing the language to indicate that the last version of `skuid-cli` to support `sfx` was `v0.5.1`.
